### PR TITLE
pluginhandler: remove prepare, build and install scriptlets

### PR DESCRIPTION
**Pull request by sergiusens**
These have long since been deprecated in favour of the
override-{step-name} syntax.

The integration tests remain to ensure backwards compatibility is in
order.

LP: #1794854

Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [ ] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: schema/snapcraft.yaml [hunk 1]
```diff
@@ -583,29 +488,13 @@ properties:
             items:
               type: string
             default: ['*']
-          snap: # FIXME: This is deprecated, replaced by `prime`, below.
```
Comment (kyrofa) [inline schema/snapcraft.yaml]: This should probably be done in its own PR, no?

File: tests/unit/lifecycle/test_lifecycle.py [hunk 2]
```diff
@@ -385,77 +385,6 @@ def test_non_prime_and_no_version(self):
         lifecycle.execute(steps.PULL, project_config)
 
 
-class DirtyBuildScriptletTestCase(LifecycleTestBase):
```
Comment (kyrofa) [inline tests/unit/lifecycle/test_lifecycle.py]: This is actually quite an important test, can you just change it to use the override scriptlets instead?

Comment (sergiusens) [reply tests/unit/lifecycle/test_lifecycle.py]: sure, I just saw old methods and decided to wipe. good catch

File: schema/snapcraft.yaml (hunk 1 above)
Comment (sergiusens) [reply schema/snapcraft.yaml]: it even has its own bug, let me fix that slip

File: tests/unit/lifecycle/test_lifecycle.py (hunk 2 above)
Comment (sergiusens) [reply tests/unit/lifecycle/test_lifecycle.py]: wait, these same tests already exist
